packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-latencies.ts

```diff
@@ -111,6 +128,62 @@ describe('internal-plugin-metrics', () => {
       assert.deepEqual(cdl.getMeetingInfoReqResp(), 10);
     });
 
+    describe('measureLatency', () => {
+      let clock;
+      let callbackStub;
+      let saveLatencySpy;
+
+      beforeEach(() => {       
+        clock = sinon.useFakeTimers(); 
+        
+        saveLatencySpy = sinon.stub(cdl, 'saveLatency');
+      });
+
+      afterEach(() => {
+        clock.restore();
+        sinon.restore();
+      });
+      
+      it('checks measureLatency with overwrite false', async () => {
+        const key = 'internal.client.pageJMT';
+        const overwrite = false;
+        callbackStub = sinon.stub().resolves();
+
+        const promise = cdl.measureLatency(callbackStub, 'internal.client.pageJMT', overwrite);
+        clock.tick(50);
+
+        await promise;
+        assert.calledOnce(callbackStub);
+        sinon.assert.calledWith(saveLatencySpy, key, 50, overwrite)       
+      });
+
+      it('checks measureLatency with overwrite true', async () => {
+        const key = 'internal.client.pageJMT';
+        const overwrite = true;
+        callbackStub = sinon.stub().resolves();
+
+        const promise = cdl.measureLatency(callbackStub, 'internal.client.pageJMT', overwrite);
+        clock.tick(50);
+
+        await promise;
+        assert.calledOnce(callbackStub);
+        sinon.assert.calledWith(saveLatencySpy, key, 50, overwrite)       
+      });
+
+      it('checks measureLatency when callBack rejects', async () => {
+        const key = 'internal.client.pageJMT';
+        const overwrite = true;
+        callbackStub = sinon.stub().rejects(new Error('some error'));
+
+        const promise = cdl.measureLatency(callbackStub, 'internal.client.pageJMT', overwrite);
+        clock.tick(50);
+
+        await await assert.isRejected(promise);
```

Comment:
issue: This should check the rejection value is the same as the rejection value used by the callback.